src/core/meshcop/tcat_agent.hpp

```diff
@@ -364,8 +393,206 @@ class TcatAgent : public InstanceLocator, private NonCopyable
 DefineCoreType(otTcatVendorInfo, MeshCoP::TcatAgent::VendorInfo);
 
 DefineMapEnum(otTcatApplicationProtocol, MeshCoP::TcatAgent::TcatApplicationProtocol);
+DefineMapEnum(otTcatDeviceIdType, MeshCoP::TcatAgent::TcatDeviceIdType);
+
+// Command class TLVs
+typedef UintTlvInfo<MeshCoP::TcatAgent::kTlvResponseWithStatus, uint8_t> ResponseWithStatusCommandTlv;
+
+/**
+ * Represent Device Type and Status
+ *
+ */
+struct DeviceTypeAndStatus
+{
+    uint8_t mRsv : 3;
+    bool    mIsCommisionned : 1;
+    bool    mThreadNetworkActive : 1;
+    bool    mIsBorderRouter : 1;
+    bool    mRxOnWhenIdle : 1;
+    bool    mDeviceType : 1;
+};
+
+// Advertisement TLVs
+
+OT_TOOL_PACKED_BEGIN
+class AdvertisementTlv : public ot::Tlv
+{
+public:
+    /**
+     * TCAT Advertisement TLV Types.
```

Comment:
In the spec it's the name "TCAT Device Property TLV".  We can adapt it here, or adapt it in the spec (maybe TCAT Advertisement TLV is better.)

Reply:
I would say let's adapt it in speck.

Reply:
Ok, I've created https://threadgroup.atlassian.net/browse/BHC-736 for this rename.

Reply:
https://threadgroup.atlassian.net/browse/BHC-736